### PR TITLE
Alerting: Send alerts to the remote Alertmanager

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR implements sending alerts to the remote Alertmanager by holding a queue in memory and flushing batches of alerts.

This is an alternative approach to https://github.com/grafana/grafana/pull/76772

Related issue: https://github.com/grafana/grafana/issues/76692

### CODE REVIEW

File: pkg/services/ngalert/remote/alertmanager.go [hunk 1]
```diff
@@ -247,10 +247,45 @@ func (am *Alertmanager) TestTemplate(ctx context.Context, c apimodels.TestTempla
...
+			am.log.Warn("Error closing response body", "err", err)
+		}
+	}()
+
+	if res.StatusCode != http.StatusOK {
+		am.log.Debug("Alertmanager readiness check unsuccessful", "statusCode", res.StatusCode)
+		return false
+	}
+
+	// Wait for active senders.
```
Comment (santihernandezc) [inline pkg/services/ngalert/remote/alertmanager.go:272]: We're gonna keep this part for the final implementation of `Alertmanager.Ready()`

File: pkg/services/ngalert/remote/alertmanager.go [hunk 2]
```diff
@@ -195,29 +215,9 @@ func (am *Alertmanager) GetAlertGroups(ctx context.Context, active, silenced, in
...
-			StartsAt:    pa.StartsAt,
-			Alert:       pa.Alert,
-		})
-	}
-
-	params := amalert.NewPostAlertsParamsWithContext(ctx).WithAlerts(alerts)
-	_, err := am.amClient.Alert.PostAlerts(params)
-	return err
+func (am *Alertmanager) PutAlerts(ctx context.Context, alerts apimodels.PostableAlerts) error {
+	am.sender.SendAlerts(alerts)
```
Comment (gotjosh) [inline pkg/services/ngalert/remote/alertmanager.go:219]:
```suggestion
    am.log.Debug("Sending alerts to a remote alertmanager", "url", am.url, "alerts", len(alerts))
	am.sender.SendAlerts(alerts)
```

File: pkg/services/ngalert/remote/alertmanager.go [hunk 3]
```diff
@@ -247,10 +247,45 @@ func (am *Alertmanager) TestTemplate(ctx context.Context, c apimodels.TestTempla
 }
 
 func (am *Alertmanager) StopAndWait() {
+	am.sender.Stop()
 }
 
+// TODO: change implementation, this is only useful for testing other methods
 func (am *Alertmanager) Ready() bool {
-	return false
+	readyURL := strings.TrimSuffix(am.url, "/") + "/-/ready"
```
Comment (gotjosh) [inline pkg/services/ngalert/remote/alertmanager.go:255]: I think we should move all of this ready code to `ApplyConfig`  and use a bool to know wether we're ready or not. 

`Ready` is called everytime we try to get an alertmanager (even on the API) so I don't think we should be making HTTP calls wether this is called.

**Review by gotjosh — COMMENTED** [review]
LGTM, but please take a look at my comments.

For anyone following along, we discussed a lot of this offline.

**Review by gotjosh — APPROVED** [review]

**Pull request merged**